sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,567 @@
+
+    Parameters
+    ==========
+    length : Sympifyable
+        A SymPy expression representing the Beam's length.
+    elastic_modulus : Sympifyable
+        A SymPy expression representing the Beam's Modulus of Elasticity.
+        It is a measure of the stiffness of the Beam material.
+    second_moment : Sympifyable
+        A SymPy expression representing the Beam's Second moment of area.
+        It is a geometrical property of an area which reflects how its
+        points are distributed with respect to its neutral axis.
+
+    Examples
+    ========
+    >>> from sympy.physics.continuum_mechanics.beam import Beam, PointLoad, DistributedLoad
+    >>> from sympy import Symbol
+    >>> E = Symbol('E')
+    >>> I = Symbol('I')
+    >>> b = Beam(4, E, I)
+    >>> Load_1 = PointLoad(location = 0, value = -3)
+    >>> Load_2 = PointLoad(location = 4, value = -9)
+    >>> Load_3 = DistributedLoad(value = 6, start = 2, order = 0)
+    >>> b.apply_loads(Load_1, Load_2, Load_3)
+    >>> b.apply_boundary_conditions(deflection = [(4, 0)])
+    >>> b.boundary_conditions
+    {'deflection': [(4, 0)], 'moment': [], 'slope': []}
+    >>> b.load_distribution()
+    -3*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*SingularityFunction(x, 4, -1)
+    >>> b.shear_force()
+    -3*SingularityFunction(x, 0, 0) + 6*SingularityFunction(x, 2, 1) - 9*SingularityFunction(x, 4, 0)
+    >>> b.bending_moment()
+    3*SingularityFunction(x, 0, 1) - 3*SingularityFunction(x, 2, 2) + 9*SingularityFunction(x, 4, 1)
+    >>> b.slope()
+    (3*SingularityFunction(x, 0, 2)/2 - SingularityFunction(x, 2, 3) + 9*SingularityFunction(x, 4, 2)/2 - 7)/(E*I)
+    >>> b.deflection()
+    (-7*x + SingularityFunction(x, 0, 3)/2 - SingularityFunction(x, 2, 4)/4 + 3*SingularityFunction(x, 4, 3)/2)/(E*I)
+    """
+
+    def __init__(self, length, elastic_modulus, second_moment):
+        self._length = length
+        self._elastic_modulus = elastic_modulus
+        self._second_moment = second_moment
+        self._boundary_conditions = {'deflection': [], 'moment': [], 'slope': []}
+        self._load = 0
+
+    def __str__(self):
+        str_sol = 'Beam(%s, %s, %s)' % (sstr(self._length), sstr(self._elastic_modulus), sstr(self._second_moment))
+        return str_sol
+
+    __repr__ = __str__
+
+    @property
+    def length(self):
+        """Length of the Beam."""
+        return self._length
+
+    @length.setter
+    def length(self, l):
+        self._length = sympify(l)
+
+    @property
+    def elastic_modulus(self):
+        """Young's Modulus of the Beam. """
+        return self._elastic_modulus
+
+    @elastic_modulus.setter
+    def elastic_modulus(self, e):
+        self._elastic_modulus = sympify(e)
+
+    @property
+    def second_moment(self):
+        """Second moment of area of the Beam. """
+        return self._second_moment
+
+    @second_moment.setter
+    def second_moment(self, i):
+        self._second_moment = sympify(i)
```

Comment:
If you want to implement an extra nice feature and have time, it would be really cool to have some kind of `CrossSection` class that constains the 2D geometry of the cross section. This class could then compute the second moment of area with a method.